Makefile

```diff
@@ -0,0 +1,30 @@
+CNT=0
+
+.PHONY: vet
+vet:
+	go vet ./...
+
+.PHONY: gfmrun
+gfmrun:
+	@while read -r line; do \
+		if [[ $$line == "package main" ]]; \
+		then \
+			c=$$(( $${c:-$(CNT)}+1 )); \
+		fi \
+	done <README.md; \
+	gfmrun -c $$c -n README.md
+
+.PHONY: toc
+toc:
+	node_modules/.bin/markdown-toc -i README.md
+	git diff --exit-code
+
+.PHONY: generate
+generate:
+	go generate ./...
+	git diff --exit-code
+
+.PHONY: test
+test:
+	go test -v --coverprofile=cli.coverprofile github.com/urfave/cli/
+	go test -v --coverprofile=altsrc.coverprofile github.com/urfave/cli/altsrc
```

Comment:
This now requires make installed, which is not available on Windows. Everyone has go installed when compiling go, why can't we have a build.go script that you go run

Reply:
the forced push was me reverting the previous commit including make. Now we are back to what we had previously. I am now working on a build.go that will take care of these things.